src/language_features/kak_lsp_custom.rs

```diff
@@ -0,0 +1,318 @@
+}
+
+/// Send the response back to Kakoune. This could be either:
+/// a) Instructions to move the cursor to the next/previous symbol.
+/// b) Instructions to show hover information (without actually moving the
+///    visible user cursor) _of_ the next/previous symbol
+fn next_or_prev_response(
+    meta: EditorMeta,
+    ctx: &mut Context,
+    maybe_details: Option<(String, KakounePosition, String, SymbolKind)>,
+    hover: bool,
+) {
+    if let Some((filename, symbol_position, name, kind)) = maybe_details {
+        if hover {
+            // Create hover `req_params` just like we do in `hover::text_document_hover()`
+            let req_params = HoverParams {
+                text_document_position_params: TextDocumentPositionParams {
+                    text_document: TextDocumentIdentifier {
+                        uri: Url::from_file_path(&meta.buffile).unwrap(),
+                    },
+                    position: get_lsp_position(&meta.buffile, &symbol_position, ctx).unwrap(),
+                },
+                work_done_progress_params: Default::default(),
+            };
+
+            // This context is shown at the top of the modal
+            let context = format!(
+                                "line {}:{}:{{+b@KindAndName}}{:?} {}{{KindAndName}} (Press `g` to goto this position. Press any other key to continue)",
+                                symbol_position.line, symbol_position.column, kind, escape_brace(&name)
+                            );
+
+            // This kak script is appended after the kakoune `lsp-show-hover`
+            let do_after = format!(
+                "on-key %[eval %sh[
+                     if [ \"$kak_key\" = \"g\" ];
+                     then echo 'info -style modal
+                                exec {}g{}lh
+                                ';
+                     else echo 'info -style modal';
+                     fi
+                 ]]",
+                symbol_position.line, symbol_position.column
+            );
+
+            // Make a HoverRequest to the backend language server and then call `hover::editor_hover()`
+            ctx.call::<HoverRequest, _>(
+                meta,
+                req_params,
+                move |ctx: &mut Context, meta, result| {
+                    editor_hover(
+                        meta,
+                        Some(HoverModal { context, do_after }),
+                        PositionParams {
+                            position: symbol_position,
+                        },
+                        result,
+                        ctx,
+                    )
+                },
+            );
+        } else {
+            let location = format!(
+                "%§{}§ {} {}",
+                filename.replace("§", "§§"),
+                symbol_position.line,
+                symbol_position.column
+            );
+            let command = format!("edit! -existing {}", location);
```

Comment:
for the unlikely case a filename starts with `-`

```suggestion
            let command = format!("edit! -existing -- {}", location);
```